### PR TITLE
fix(android/engine): Fix toHex() for null string

### LINKED ISSUE #4990: bug(android/engine): TypeError: Cannot read property 'length' of null
Sentry Issue: [KEYMAN-WEB-21](https://sentry.keyman.com/organizations/keyman/issues/4829/?referrer=github_integration)

```
TypeError: Cannot read property 'length' of null
  at toHex (/data/user/0/com.tavultesoft.kmapro/app_data/keyboard.html:349:35)
  at oskCreateKeyPreview (/data/user/0/com.tavultesoft.kmapro/app_data/keyboard.html:259:101)
  at showKeyTip (source/kmwembedded.ts:94:11)
  at highlightKey (source/osk/visualKeyboard.ts:1944:9)
  at a.<anonymous> (source/osk/visualKeyboard.ts:1173:7)
```

From keyboard.html
```javascript
    function toHex(theString) {
      var hexString = '';
      for (var i=0; i < theString.length; i++) {  // <----- length of null
        var theHex = theString.charCodeAt(i).toString(16).toUpperCase();
        while (theHex.length < 4) {
          theHex = '0' + theHex;
        }
        theHex = '\\u' + theHex;
        hexString += theHex;
      }
      return hexString;
    }
```

Steps for repro
1. In Keyman for Android, install hieroglyphic keyboard
2. On the hieroglyphic default layer, press the blank key on the bottom right
hieroglyphic default layer

3. The null string causes the exception in keyboard.html

### PR DESCRIPTION
**Pull request by darcywong00**
Fixes #4990 by checking for null string in toHex()

* Also as a chore, update the display language list in whatsnew.md

### CODE REVIEW

**Review by mcdurdin — APPROVED** [review]
I tend to prefer preconditions:

```js
if(theString == null) return '';
```

Especially helps if there are many preconditions.

Question in my mind why the caller is passing `null` to `toHex()` instead of `''`? That's probably also worth checking out.

But LGTM :)

**Comment by darcywong00** [comment]
> Question in my mind why the caller is passing `null` to `toHex()` instead of `''`?

In this case, it's from oskCreatePopup when no keycap is defined.
```
          if(typeof(obj[i].text) != 'undefined' && obj[i].text != '') s=s+':'+toHex(obj[i].text);
```

Maybe that needs a null check?

**Comment by mcdurdin** [comment]
> Maybe that needs a null check?

Yes please. Leave the current fix in as well though.

File: android/KMEA/app/src/main/assets/keyboard.html [hunk 1]
```diff
@@ -345,16 +345,17 @@
     }
 
     function toHex(theString) {
+      if (theString == null) {
```
Comment (mcdurdin) [inline android/KMEA/app/src/main/assets/keyboard.html:348]:
```suggestion
      if (typeof theString != 'string') {
```

**Review by mcdurdin — APPROVED** [review]

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.38-alpha](https://keyman.com/downloads/releases/alpha/15.0.38)

**Pull request merged**